helix-core/src/test/java/org/apache/helix/monitoring/mbeans/TestResourceMonitor.java

```diff
@@ -217,10 +217,13 @@ public void testReportData() throws JMException {
     monitor.setRebalanceState(ResourceMonitor.RebalanceStatus.INTERMEDIATE_STATE_CAL_FAILED);
     Assert.assertEquals(monitor.getRebalanceState(),
         ResourceMonitor.RebalanceStatus.INTERMEDIATE_STATE_CAL_FAILED.name());
+
+    // Has to unregister this monitor to clean up. Otherwise, later tests may be affected and fail.
+    monitor.unregister();
```

Comment:
In this case, put it in finally block and check if monitor is null before unregister.

Reply:
Done.